qa/rpc-tests/test_framework/blocktools.py

```diff
@@ -30,6 +30,12 @@ def create_block(hashprev, coinbase, nTime=None, txns=None, ctor=True):
     block.calc_sha256()
     return block
 
+def make_conform_to_ctor(block):
+    for tx in block.vtx:
+        tx.rehash()
```

Comment:
why do we need to `rehash()`?

Reply:
Because transactions may have modifications, but we can't trust that the  cached hash value is updated. A better CTransaction API would keep track of dirty cache for us.

Reply:
ok I guess it makes sense